mypy/messages.py

```diff
@@ -1025,14 +1025,10 @@ def does_not_return_value(self, callee_type: Type | None, context: Context) -> N
         callee_type = get_proper_type(callee_type)
         if isinstance(callee_type, FunctionLike):
             name = callable_name(callee_type)
-        if name is not None:
-            self.fail(
-                f"{capitalize(name)} does not return a value",
-                context,
-                code=codes.FUNC_RETURNS_VALUE,
-            )
-        else:
-            self.fail("Function does not return a value", context, code=codes.FUNC_RETURNS_VALUE)
+        message = "{} does not return a value (or returns None)".format(
+            "Function" if name is None else capitalize(name)
+        )
```

Comment:
```suggestion
        name = capitalize(name) if name else "Function"
        message = f"{name} does not return a value (or returns None)"
```

Reply:
```suggestion
        name = "Function" if name is None else capitalize(name)
        message = f"{name} does not return a value (or returns None)"
```

Thanks for the suggestion! Maybe an explicit check for `None`?

Reply:
Of course. :)

Reply:
(I "simplified" it because an empty string wouldn't make for a good interpolation either, but it also suggests it's a possibility where currently it's not.)

Reply:
Considering the implementation of `callable_name()` it seems to me that we could omit the capitalization:

https://github.com/python/mypy/blob/b49be105d2940e3a0607f5ec76f519931b0d0a08/mypy/messages.py#L2963-L2967

It always returns either `None`, `"Foo"`, or `<Foo>`, where capitalization does not seem to be applicable. What do you think?

Reply:
Addressed in dceaac1c350ba05f89366360dbb3e36cb4817185

Reply:
Looks much better!